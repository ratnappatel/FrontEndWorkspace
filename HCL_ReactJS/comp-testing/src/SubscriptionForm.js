import React,{useState} from 'react'

export default function SubscriptionForm() {

    const [isDisabled, setIsDisabled]=useState(true);
    const[email,setEmail]=useState("");


   function handleChange(e){
    setEmail(e.target.value);
    setIsDisabled(e.target.value=== "");
   }

  return (
    <div>
        <h1>Subscription Form</h1>
        <form>
            <label htmlFor="email"> Email Address</label>
            <input type="email" id="email" placeholder='Email Address' value={email}
                onChange={handleChange}/><br></br>
            <input type="checkbox" name="agreement" id="agreement"/>
            <label htmlFor="agreement">I agree to all Terms & Conditions</label><br></br>
            <button name="subscribe" type="submit" disabled={isDisabled}>Subscribe</button>
        </form>
    </div>
  )
}
