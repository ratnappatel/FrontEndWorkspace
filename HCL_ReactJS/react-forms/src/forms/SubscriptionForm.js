import React, { Component } from 'react'

export default class SubscriptionForm extends Component {
  render() {
    return (
      <div>
        <h1>SubscriptionForm [Uncontrolled Form]</h1>
        <form>
            
        </form>
      </div>
    )
  }
}
