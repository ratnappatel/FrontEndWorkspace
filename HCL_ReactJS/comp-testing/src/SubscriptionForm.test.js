import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Renderer } from 'react-test-renderer';
import SubscriptionForm from './SubscriptionForm';


it('Check Subscription Button must be Disabled',()=>{
    render(<SubscriptionForm/>)
    expect(screen.getByRole("button",{name: /subscribe/i})).toBeDisabled();
});

it('The Subscribe Button becomes enabled when onChange event occured..',()=>{
    render(<SubscriptionForm/>);
    userEvent.type(screen.getByRole('textbox',{name: /email/i}),'abc@gmail.com');
    expect(screen.getByRole("button",{name: /subscribe/i})).toBeEnabled();
})

it('Test to match snapshot of component',()=>{
    const subFormTree= Renderer.create(<SubscriptionForm/>).toJSON();

    expect(subFormTree).toMatchSnapshot();
})