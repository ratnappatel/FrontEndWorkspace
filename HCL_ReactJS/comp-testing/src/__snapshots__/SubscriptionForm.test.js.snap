// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to match snapshot of component 1`] = `
<div>
  <h1>
    Subscription Form
  </h1>
  <form>
    <label
      htmlFor="email"
    >
       Email Address
    </label>
    <input
      id="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="email"
      value=""
    />
    <br />
    <input
      id="agreement"
      name="agreement"
      type="checkbox"
    />
    <label
      htmlFor="agreement"
    >
      I agree to all Terms & Conditions
    </label>
    <br />
    <button
      disabled={true}
      name="subscribe"
      type="submit"
    >
      Subscribe
    </button>
  </form>
</div>
`;
